src/components/SearchHeader/SearchHeader.tsx

```diff
@@ -5,60 +5,67 @@ import styles from './SearchHeader.module.css';
+  const [input, setInput] = useState<string>('');
 
   function handleEnter(e: any): void {
     if (e.key === 'Enter') {
-      history.push('/search', {searchInput: input})
+      history.push('/search', { searchInput: input });
     }
   }
 
   return (
     <>
-      <div className={styles["search-header"]}>
-        <div className={styles["search-header-top"]}>
-          <div className={styles["digital-kampung-div"]}>
+      <div className={styles['search-header']}>
+        <div className={styles['search-header-top']}>
+          <div className={styles['digital-kampung-div']}>
             <Link
-              style={{textDecoration: 'none', color: "black"}}
+              style={{ textDecoration: 'none', color: 'black' }}
               to={{
-                pathname: "/"
+                pathname: '/',
               }}
             >
-              <img className={styles["digital-kampung-icon"]} src={darkLogo} alt=""></img>
+              <img className={styles['digital-kampung-icon']} src={darkLogo} alt=""></img>
             </Link>
           </div>
           <div>
-            <div className={styles["header-button-group"]}>
-              <Button basic className={styles["header-button"]}>kampung Centre</Button>
-              <Button basic className={styles["header-button"]}>Sign up</Button>
-              <Button className={styles["header-button-primary"]}>Log in</Button>
+            <div className={styles['header-button-group']}>
+              <Button basic className={styles['header-button']}>
+                kampung Centre
+              </Button>
+              <Button basic className={styles['header-button']}>
+                Sign up
+              </Button>
+              <Button className={styles['header-button-primary']}>Log in</Button>
             </div>
-         </div>
+          </div>
         </div>
-        <div className={styles["search-bar"]}>
-          <Input 
-            fluid 
-            className={styles["search-bar-input"]} 
-            placeholder='search for hawker centre / food' 
-            value={input} 
-            onChange={e => setInput(e.target.value)}
+        <div className={styles['search-bar']}>
+          <Input
+            fluid
+            className={styles['search-bar-input']}
+            placeholder="search for hawker centre / food"
+            value={input}
+            onChange={(e) => setInput(e.target.value)}
             onKeyPress={handleEnter}
           />
-          <Link to={{
-            pathname: "/search",
-            state: {
-              searchInput: input,
-            }
-          }}>
-            <Button id="search" className={styles["search-button-primary"]}>Search</Button>
+          <Link
+            to={{
+              pathname: '/search',
+              state: {
+                searchInput: input,
+              },
+            }}
+          >
+            <Button id="search" className={styles['search-button-primary']}>
```

Comment:
The button's height is slightly less than the input bar because it is wrapped in a link. Either specify the height in css or remove the wrap (I would recommend using the normal `onClick` handler and use `history.push`).

Reply:
If we are shifting towards placing the search query in the query parameters then yeah, I think you're right. That's a better idea.